shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py

```diff
@@ -231,12 +232,33 @@ def find_all_local_file_names(source_folder_name: str = None) -> list:
     list: A list of all files that exist in the current working directory.
     """
     logger.debug(f"Finding all local file names in {source_folder_name}...")
-    cwd = os.getcwd()
-    cwd_extension = os.path.normpath(f"{cwd}/{source_folder_name}/**")
-    all_paths = glob.glob(cwd_extension, recursive=True)
-    logger.debug(f"Found {len(all_paths)}.")
-    logger.debug(all_paths)
-    return remove_directories_from_path_list(all_paths)
+    # cwd = os.getcwd()
+    # if not source_folder_name:
+    #     search_path = os.getcwd()
+    # else:
+    #     search_path = os.path.normpath(f"{cwd}/{source_folder_name}/**")
+    # all_paths = glob.glob(search_path, recursive=True)
+    # print(all_paths)
+    # logger.debug(f"Found {len(all_paths)}.")
+    # logger.debug(all_paths)
+    # return remove_directories_from_path_list(all_paths)
+    if source_folder_name is None:
+        source_folder_name = (
+            os.getcwd()
+        )  # Use current working directory if no directory provided
+    try:
+        file_list = [
+            os.path.join(source_folder_name, filename)
```

Comment:
the original script looks like it intentionally removed the source folder. did we test to make sure this doesn't break `find_matching_files`

Reply:
Also `os.path.join(source_folder_name, filename)` is being used twice here. I recommend using a variable name.

Reply:
personally the following is more readable but I don't have a strong opinion other than using a var name for the full path assuming the full path doesn't break the other functions using this method.

```
all_files(pick a better var name) = os.listdir(source_folder_name)
file_list=[]
for file in all_files:
    full_path  = os.path.join(source_folder_name,file)
    if os.path.isfile(full_path):
        file_list.append(full_path
```